store/mongo/datastore_mongo.go

```diff
@@ -710,7 +710,6 @@ func (db *DataStoreMongo) IsArtifactUnique(ctx context.Context,
 		} else if len(i.ArtifactMeta.Depends) == 0 {
 			return false, nil
 		}
-		return false, err
```

Comment:
This code is a bit foreign to me, so apologies if I get this wrong, but I *think* this might not be correct. If you explicitly check for only one dependency count, that means that if you upload an artifact that has two dependencies, you essentially throw the whole uniqueness criterion out, and you can upload whatever you want. Including two artifacts that have identical depends information, as long as they have at least two entries.

For consistency I think we should apply the same constraints in the OS version as in the Enterprise version, even though we won't actually use the depends information. This could become important if people upgrade, for example.

Reply:
I know this looks like it violates the conflicting depends criterion, but this function is only used by the generate artifact endpoint where you can't specify any additional depends besides device type. It's not a pretty fix, but it's what's currently in the enterprise repo (that return line is just a glitch that I mistakenly made it in merging my last PR). We should probably follow up with creating a task adding an optional depends parameter to the generate artifact request and get this in the above function so it doesn't look that confusing.

Reply:
Ah, that explains it!